test/modules/standard/memory/countMemory/countMemory.makegood

```diff
@@ -28,33 +28,37 @@ $realMemInBytesVal  = $realMemInKBytesVal * 1024;
 $realMemInMBytesVal = $realMemInKBytesVal / 1024;
 $realMemInGBytesVal = $realMemInMBytesVal / 1024;
 
-$realMemInKBytes = sprintf("%g", $realMemInKBytesVal);
-$realMemInBytes  = sprintf("%g", $realMemInBytesVal);
-$realMemInMBytes = sprintf("%g", $realMemInMBytesVal);
-$realMemInGBytes = sprintf("%g", $realMemInGBytesVal);
+$realMemInBytes  = print_chapel_real($realMemInBytesVal);
+$realMemInKBytes = print_chapel_real($realMemInKBytesVal);
+$realMemInMBytes = print_chapel_real($realMemInMBytesVal);
+$realMemInGBytes = print_chapel_real($realMemInGBytesVal);
 			   
-
-#
-# if a string is all digits, add a trailing .0 to match
-# Chapel's default formatting of reals
-#
-if ($realMemInKBytes =~ m/^\d+$/) {
-    $realMemInKBytes .= ".0";
-}
-if ($realMemInBytes =~ m/^\d+$/) {
-    $realMemInBytes .= ".0";
-}
-if ($realMemInMBytes =~ m/^\d+$/) {
-    $realMemInMBytes .= ".0";
-}
-if ($realMemInGBytes =~ m/^\d+$/) {
-    $realMemInGBytes .= ".0";
-}
-
-
 printf "memory is: %s\n", $realMemInBytes;
 printf " in bytes: %s\n", $realMemInBytes;
 printf "    in kb: %s\n", $realMemInKBytes;
 printf "    in mb: %s\n", $realMemInMBytes;
 printf "    in gb: %s\n", $realMemInGBytes;
 
+#
+# This is for change the formatting of a float number
+# to the Chapel's formatting of reals
+#
+sub print_chapel_real{
+	$retVal = 0;
+	if( ($_[0] >= 100000.0) and ($_[0] < 1000000.0) ){
+		$retVal = sprintf("%.5e", $_[0]);
+	}
+	else {
+		$retVal = sprintf("%g", $_[0]);
+	}
+
+	#
+	# if a string is all digits, add a trailing .0 to match
+	# Chapel's default formatting of reals
+	#
+	if ($retVal =~ m/^\d+$/) {
+	    $retVal .= ".0";
+	}
+
+	return $retVal;
+}
```

Comment:
@mppf , please confirm this resolution of the merge conflict (compare to #3870) 
FYI, when the cherry-pick was done, this is the conflict git left in the file (below)- I resolved it in favor of the incoming patch. 

    26	$realMemInKBytesVal = $memoryInKBytes;
    27	$realMemInBytesVal  = $realMemInKBytesVal * 1024;
    28	$realMemInMBytesVal = $realMemInKBytesVal / 1024;
    29	$realMemInGBytesVal = $realMemInMBytesVal / 1024;
    30	
    31	<<<<<<< HEAD
    32	$realMemInKBytes = sprintf("%g", $realMemInKBytesVal);
    33	$realMemInBytes  = sprintf("%g", $realMemInBytesVal);
    34	$realMemInMBytes = sprintf("%g", $realMemInMBytesVal);
    35	$realMemInGBytes = sprintf("%g", $realMemInGBytesVal);
    36				   
    37	
    38	#
    39	# if a string is all digits, add a trailing .0 to match
    40	# Chapel's default formatting of reals
    41	#
    42	if ($realMemInKBytes =~ m/^\d+$/) {
    43	    $realMemInKBytes .= ".0";
    44	}
    45	if ($realMemInBytes =~ m/^\d+$/) {
    46	    $realMemInBytes .= ".0";
    47	}
    48	if ($realMemInMBytes =~ m/^\d+$/) {
    49	    $realMemInMBytes .= ".0";
    50	}
    51	if ($realMemInGBytes =~ m/^\d+$/) {
    52	    $realMemInGBytes .= ".0";
    53	}
    54	
    55	
    56	=======
    57	$realMemInBytes  = print_chapel_real($realMemInBytesVal);
    58	$realMemInKBytes = print_chapel_real($realMemInKBytesVal);
    59	$realMemInMBytes = print_chapel_real($realMemInMBytesVal);
    60	$realMemInGBytes = print_chapel_real($realMemInGBytesVal);
    61				   
    62	>>>>>>> daa9e65... Test updated for the new real format
    63	printf "memory is: %s\n", $realMemInBytes;
    64	printf " in bytes: %s\n", $realMemInBytes;
    65	printf "    in kb: %s\n", $realMemInKBytes;
    66	printf "    in mb: %s\n", $realMemInMBytes;
    67	printf "    in gb: %s\n", $realMemInGBytes;